File: ASSEMBLER_vs_C_vs_CPP_vs_JS/anObject.js
main();

function main(){

var myObject = {
	name:"Hello World",
	compare : function(a,b){var res = (a>b)? a :  b;return res;}
    /* this is the same as:
    function(a,b){if(a>b){return a;}else{return b;}}
         or
     function(a,b){if(a>b) return a; else return b;}
        */
	};
// or use: new Object();
// and fill the variables like this
// myObject.name = "Hello World";
// myObject.compare = function(a,b){if(a>b){return a;}else{return b;}};

var a = 23, b = 5;

alert("The Object named: "+ 
	myObject.name +"\nCompared: "
	+ a +" with "+ b +
	"\n" + myObject.compare(a,b) 
	+ " is bigger\n");
myObject = null;
return 0;

}

File: ASSEMBLER_vs_C_vs_CPP_vs_JS/anObject.c
// #include<string.h>
#include<stdio.h>



compare(int a, int b){

	if(a>b){return a;}else{return b;}
}

int main(void)
{
	char name[11] = "Hello World";
	int a = 23,b = 5;
	printf("The Object named: %s\nCompared: %d with %d.\n%d is bigger!\n",name,a,b,compare(a,b));
	// printf("\n);
	// printf("%d with %d\n",a,b);
	// printf("%d is bigger\n",compare(a,b));
	return 0;
}